Integration of Milvus database into camel-agent

MansonMY opened the pull request:
## Description

This code is a take-home exam, which is for the integration of the Milvus database into the camel agent. This work took references 
mainly from the coding work by established qdrant vector database in camal/storages and milvus sites.

Wendong-Fan commented:
Hey @MansonMY , thanks for the PR, is it ready for review or you will do more enhancement further?

MansonMY commented:
> Hey @MansonMY , thanks for the PR, is it ready for review or you will do more enhancement further?

Hey, @Wendong-Fan , Thanks for the comments. I double-checked the above code again and made a few changes. I think it is ready to be reviewed.

The further enhancement parts I will mainly focus on adding some parts about exception handling, such as what to do if data fails to be inserted, or what to do if a deletion fails. Additionally, I will check if the methods in the base class have been overridden; if not, it might also be necessary to consider whether they need to be overridden and how to do so.

Before to do further enhancements, please let me know your comments on the current version and if you are happy with them. Thanks

Wendong-Fan commented on camel/storages/vectordb_storages/milvusDB_v2.py:
```diff
@@ -0,0 +1,99 @@
+from pymilvus import MilvusClient, Collection
+from typing import Any, Dict, List, Optional, Tuple, Union, cast
```
`Dict`,`Optional`,`Union`,`cast` not used

Wendong-Fan commented on camel/storages/vectordb_storages/milvusDB_v2.py:
```diff
@@ -0,0 +1,99 @@
+from pymilvus import MilvusClient, Collection
```
better to move the import into class

MansonMY replied:
I have moved it into the function, Thanks

Wendong-Fan commented on camel/storages/vectordb_storages/milvusDB_v2.py:
```diff
@@ -0,0 +1,99 @@
+    # Create a new collection in Milvus with a unique name and vector dimensions
+    def create_collection(self, **kwargs: Any,) -> str:
+        try:    
+            collection_name  = self.generate_collection_name()
```
we should also allow the user to define the collection name

Wendong-Fan commented on camel/storages/vectordb_storages/milvusDB_v2.py:
```diff
@@ -0,0 +1,99 @@
+        self._client.drop_collection(collection_name=collection_name)
+
+    # Generate a unique collection name using UUID
+    def generate_collection_name(self) -> str:
```
this is an internal function, use `def _generate_collection_name` instead

MansonMY replied:
That is a good idea, I have changed some function into private function, please let me know your comments once you check my modified version

Wendong-Fan commented on camel/storages/vectordb_storages/milvusDB_v2.py:
```diff
@@ -0,0 +1,99 @@
+    def create_collection(self, **kwargs: Any,) -> str:
+        try:    
+            collection_name  = self.generate_collection_name()
+            self._client.create_collection(collection_name = collection_name, dimension = self.vector_dim, **kwargs,)
```
need `index_params` define metric_type and index_type,  `schema`  define metadata when creating collection

MansonMY replied:
I updated this two parameters (index_params and schema) in my new version

Wendong-Fan commented on camel/storages/vectordb_storages/milvusDB_v2.py:
```diff
@@ -0,0 +1,99 @@
+
+    # Generate a unique collection name using UUID
+    def generate_collection_name(self) -> str:
+        return str(uuid.uuid4())
```
the format of `str(uuid.uuid4())` is not acceptable as Milvus's collection name

MansonMY replied:
Thanks for the comments, I converted its format so that Milvus can collect it

Wendong-Fan commented on camel/storages/vectordb_storages/milvusDB_v2.py:
```diff
@@ -0,0 +1,99 @@
+        try:
+            vectors = [record.vector for record in records]
+            ids = [record.id for record in records] if hasattr(records[0], 'id') else None
+            self._client.insert(collection_name=self.collection_name, entities=vectors, ids=ids, **kwargs)
```
how did this work? `.insert` doesn't accpet entities and ids as input parameter

MansonMY replied:
In my latest modified script, I remodified this function and redefined the parameters，Thanks

Wendong-Fan reviewed (changes requested):
Thanks for the PR, upon review, it appears that the code is currently not functioning as expected, and there seem to be several bugs present. I would recommend conducting thorough testing(at least try to run the code yourself even didn't add unit tests) to ensure that all functions are operational before proceeding. Code format and docstring can also be further polished.

Wendong-Fan commented:
@lightaime

MansonMY commented:
Hey, @Wendong-Fan  and @lightaime , Thank you for your review. I updated my code following your comments in these two days and added the test script after the MilvusStorage module. It can run well on my local PC. I just pushed my latesd code by github.   Please don't hesitate to ask me if you have any questions. Thanks
